amd/src/edit_question.js

```diff
@@ -31,42 +31,42 @@ import {favouritePackage} from 'qtype_questionpy/utils';
  * @param {boolean} selected
  */
 export function initActionButton(card, selected) {
-    const packageHash = document.querySelector('input[name="qpy_package_hash"]');
     const packageSelected = document.querySelector('input[name="qpy_package_selected"]');
 
     if (selected) {
-        const packageFile = document.querySelector('input[name="qpy_package_file"]');
-
         // Initialize the button to change the package.
         const changeButton = card.getElementsByClassName("qpy-version-selection-button")[0];
         changeButton.addEventListener("click", (e) => {
             e.preventDefault();
 
-            packageSelected.value = false;
-            packageSelected.removeAttribute("disabled");
-
-            if (packageFile) {
-                // We want to prevent, that the same draft id will be used when changing a package.
-                packageFile.disabled = true;
-            }
-            if (packageHash) {
-                packageHash.value = '';
+            // We want to reduce the amount of unnecessary data exchange.
+            const qpyElements = document.querySelectorAll('[name^="qpy_"]:not(input[name="qpy_package_source"])');
+            for (const qpyElement of qpyElements) {
+                qpyElement.disabled = true;
             }
```

Comment:
Was passiert, wenn ich ein Paket auswähle, die Einstellungen bearbeite, dann auf den Change Button klicke und in der Liste von Paketen wieder dasselbe Paket auswähle? Sind meine Einstellungen dann weg? Gibt es eine Warnung, wenn ich auf Change klicke, dass dabei meine Einstellungen verloren gehen?

Reply:
Die Einstellungen sind, wie vorher, weg und es gibt keine Warnung.
Über `e.target.closest("form").dataset.formDirty` kann kontrolliert werden, ob die Form dirty ist und dann könnten wir auch eine Warnung ausgeben.